### ISSUE TITLE
Problem in training auto encoder for time series forecasting in keras

### ISSUE THREAD

**Issue by nis192** [keras-team/keras#10098]
I am trying to use Auto Encoder to reduce the dimension of my input time series and predict the target time series. Here is my code
#Train_X is 1000(observation)by24(timesteps)by17(dimension)
Input_=Input(batch_shape=(BATCH_SIZE,timesteps,Train_X.shape[2]))
Encoder_1 = GRU(50,dropout=0.1,unroll=Truestateful = False,return_sequences=True)(Input_)
Endoder_1_Dropout=Dropout(0.1)(Encoder_1)
Encoder_2 = GRU(10,  dropout=0.1,unroll=True,stateful = False,return_sequences=False)(Endoder_1_Dropout)
Endoder_2_Dropout=Dropout(0.1)(Encoder_2)
Decoder_1 = RepeatVector(timesteps)(Endoder_2_Dropout)
Decoder_2 = GRU(50,  dropout=0.1,unroll=True,sstateful = False,return_sequences=True)(Decoder_1)
Decoder_2_Dropout=Dropout(0.1)(Decoder_2)
Decoder_4 = GRU(Train_X.shape[2],unroll=True,stateful = False,return_sequences=True)(Decoder_2_Dropout)
AE=Model(Input_,Decoder_4)
Encoder = Model(Input_, Endoder_2_Dropout)
AE.compile(optimizer='Adadelta',loss='mean_squared_error')
AE.fit(Train_X,Train_X,epochs=10,batch_size=BATCH_SIZE,shuffle=False,validation_split=VALIDATION_SPLI)

During the training autoencoder, the error does not change from one Epoch to another. The error (cost) is around 0.024. Can you give me some advice on how can I enhance the performance of the training? The output of the Encoder stage is (BATCH_SIZE by 10). This 10 is the reduced dimension of the original time series with the dimension of 17? How can I get the timesteps? Can I use the RepeatVector as used in the following code to make the timesteps?

Extra = RepeatVector(timesteps)(Endoder_2_Dropout)
lstm_1 =  GRU(70,stateful = True,dropout=0.15,return_sequences=False,unroll=True,activation='relu',activity_regularizer=regularizers.l1(10e-6))(Extra )
predictions=Dense(1,activation='tanh')(lstm_1)
model=Model([Input_],predictions)
model.compile(loss=custom_objective, optimizer='Adadelta')
for i in range(1,10):
    history=model_1.fit([Train_X], Train_Y,
              batch_size=BATCH_SIZE, epochs=1,
              verbose=VERBOSE, shuffle=False,validation_split=VALIDATION_SPLIT)
    model.reset_states()

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
